tests/functional/floorplan_reconstruction/get-slot-test.py

```diff
@@ -0,0 +1,70 @@
+# ruff: noqa: INP001
+
+__copyright__ = """
+Copyright (c) 2025 RapidStream Design Automation, Inc. and contributors.
+All rights reserved. The contributor(s) of this file has/have agreed to the
+RapidStream Contributor License Agreement.
+"""
+import json
+import logging
+import os
+
+import pytest
+from python.runfiles import Runfiles
+
+from tapa.common.graph import Graph
+
+GRAPH_PATH = "_main/tests/functional/floorplan_reconstruction/graph/{}-graph.json"
+GOLDEN_PATH = "_main/tests/functional/floorplan_reconstruction/golden/{}-golden.json"
+SLOT_INFO_PATH = "_main/tests/functional/floorplan_reconstruction/slot/{}-slot.json"
+NEW_GOLDEN_FILE = "{}-new-golden.json"
+
+_logger = logging.getLogger(__name__)
+
+
+def test_abgraph(request: pytest.FixtureRequest) -> None:
+    """Test if the generated slot matches the golden.
+
+    If new_golden is needed, set it to true. The new golden file will be
+    generated in the bazel sandbox tmp directory. Manually updated the
+    golden file to the test directory.
+    """
+    new_golden = False
+    test_name = request.config.getoption("--test")
```

Comment:
Do you really need to pass the test name from command line? It seems appropriate to make this a parameterized test with each test_name being a parameter.

```py
@pytest.mark.parametrize("test_name", ["vadd"])
def test_abgraph(test_name: str) -> None:
```

It's easy to select the test(s) at runtime using `pytest -k`, or with bazel,

```sh
 bazel run //tapa/common:graph_test --test_arg=-k --test_arg=vadd
```